src/services/funcService.ts

```diff
@@ -0,0 +1,132 @@
+import yaml from "js-yaml";
+import rimraf from "rimraf";
+import Serverless from "serverless";
+import httpBinding from "../plugins/func/bindingTemplates/http.json";
+import { BaseService } from "./baseService";
+import fs from "fs";
+
+export class FuncService extends BaseService {
+  public constructor(serverless: Serverless, options: Serverless.Options){
+    super(serverless, options, false);
+  }
+
+  public add() {
+    const functionName = this.options["name"];
+    if (!functionName) {
+      this.log("Need to provide a name of function to add");
+      return;
+    }
+    if (this.exists(functionName)) {
+      this.serverless.cli.log(`Function ${functionName} already exists`);
+      return;
+    }
+    this.createHandler(functionName);
+    this.addToServerlessYml(functionName);
+  }
+
+  public remove() {
+    const functionName = this.options["name"];
+    if (!functionName) {
+      this.log("Need to provide a name of function to remove");
+      return;
+    }
+    if (!this.exists(functionName)) {
+      this.log(`Function ${functionName} does not exist`);
+      return;
+    }
+    const fileName = `${functionName}.js`;
+    if (fs.existsSync(fileName)) {
+      fs.unlinkSync(fileName);
+    }
+    if (fs.existsSync(functionName)) {
+      rimraf.sync(functionName);
+    }
+    this.removeFromServerlessYml(functionName);
+  }
+
+  private exists(functionName: string) {
+    return (functionName in this.slsFunctions());
+  }
+
+  private createHandler(functionName: string) {
+    this.serverless.utils.writeFileSync(`./${functionName}.js`, this.getFunctionHandler(functionName))
+  }
+
+  private addToServerlessYml(functionName: string) {
+    const functions = this.slsFunctions();
+    functions[functionName] = this.getFunctionSlsObject(functionName)
+    this.updateFunctionsYml(functions)
+  }
+
+  private removeFromServerlessYml(functionName: string) {
+    const functions = this.slsFunctions();
+    delete functions[functionName];
+    this.updateFunctionsYml(functions)
+  }
+
+  private getServerlessYml() {
+    return this.serverless.utils.readFileSync("serverless.yml");
+  }
+
+  private updateFunctionsYml(functionYml: any) {
+    const serverlessYml = this.getServerlessYml();
+    serverlessYml["functions"] = functionYml;
+    this.serverless.utils.writeFileSync("serverless.yml", yaml.dump(serverlessYml));
+  }
+
+  private getFunctionHandler(name: string) {
+    return `"use strict";
+
+module.exports.handler = async function (context, req) {
```

Comment:
I almost want to suggest making this a separate template/file (also easier to tell if it's valid code), but given that it's encapsulated here and relatively short, I'm okay with it.

Reply:
I'd love to capture this in a template file, and we actually discussed this in a previous PR (previous PR got a little murky with all the refactoring, so I closed and opened a new one). The issue I ran into was deciding what kind of file to store the template in:

```
1. .ts - transpiles to messy .js in build
2. .js - can't be included with current tsconfig (`allowJs` and `declarations` attribute cannot both be true)
3. .txt - Would need to copy to file post-build to the lib directory - seems heavy handed for this ask
```
If you have any other thoughts here, I'm open to any and all suggestions :)

Reply:
yeah, good call. I was thinking some way of reading from a file, but it gets messy in the context of the rest of the build. Will keep thinking about it -- would be nice to have separate, but we can call it good until something else better presents itself.